src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -143,49 +145,61 @@ internal string CreateCacheKey()
 
         internal string CreateCacheKey(CachedVaryBy varyBy)
         {
-            var request = HttpContext.Request;
-            var builder = new StringBuilder()
-                .Append(request.Method.ToUpperInvariant())
-                .Append(";")
-                .Append(request.Path.Value.ToUpperInvariant());
+            var request = _httpContext.Request;
+            var builder = _builderPool?.Get() ?? new StringBuilder();
```

Comment:
Can't really be null in production. I set it to null in the tests though. https://github.com/aspnet/ResponseCaching/pull/20/files/3ea31873704d140183cb4f09bcd1b615f110591b#diff-ade4e3024ab6def5a1a9db255f083dd4R659